frontend/pkg/frontend/middleware_locksubscription.go

```
@@ -40,12 +37,8 @@ func MiddlewareLockSubscription(w http.ResponseWriter, r *http.Request, next htt
 
 	subscriptionID := r.PathValue(PathSegmentSubscriptionID)
 
-	switch r.Method {
-	case http.MethodGet, http.MethodHead:
-		// These methods are read-only and don't require locking.
-	default:
-		lockClient = dbClient.GetLockClient()
-	}
+	// This may be nil when running "go test".
+	lockClient := dbClient.GetLockClient()
```

future-us should make this panic on missing dbclient.